### PR TITLE
util: add SubDuration

### PR DESCRIPTION
**Pull request by hawkingrei**
add SubDuration for mysql SUBTIME function

### CODE REVIEW

**Comment by hawkingrei** [comment]
@XuHuaiyu  PTAL

File: util/types/time.go [hunk 1]
```diff
@@ -689,6 +689,21 @@ func (d Duration) Add(v Duration) (Duration, error) {
...
 }
 
+// Sub subtracts d to d, returns a duration value.
+func (d Duration) Sub(v Duration) (Duration, error) {
+	if &v == nil {
+		return d, nil
+	}
+	dsum, err := SubInt64(int64(d.Duration), int64(v.Duration))
+	if err != nil {
+		return Duration{}, err
```
Comment (shenli) [inline util/types/time.go]: errors.Trace(err)

**Comment by shenli** [comment]
LGTM
@XuHuaiyu @tiancaiamao PTAL

**Comment by hawkingrei** [comment]
@XuHuaiyu @tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**